src/routes/Flashcards.js

```diff
@@ -6,10 +6,11 @@ const {
     getFlashcard,
     createFlashcard,
     updateFlashcard,
-    deleteFlashcard
+    deleteFlashcard,
+    getDeckWithFlashcards,
 } = require('../controllers/Flashcards');
 
 router.route('/').post(createFlashcard).get(getUserFlashcards);
-router.route('/:id').get(getFlashcard).delete(deleteFlashcard).patch(updateFlashcard);
+router.route('/:id').get([getFlashcard, getDeckWithFlashcards]).delete(deleteFlashcard).patch(updateFlashcard);
```

Comment:
[Functional Error]
If you define the functions in this way, it calls getFlashcard then getDeckWithFlashcards or calls it if getFlashcard calls next(). This isn't the intended behavior.

The user of the API needs to specify whether the id is for a deck or a flashcard.

As it stands, you may instead want to move this to the deck controller and create a separate route on the deck routes off `/deck/:id/flashcards`

Reply:
I moved this route to the Decks routes, also moved its controller to the Decks controllers. I will test it now...